contracts/ColuLocalCurrency.sol

```diff
@@ -12,21 +12,33 @@ contract ColuLocalCurrency is Ownable, Standard677Token, TokenHolder {
     string public name;
     string public symbol;
     uint8 public decimals;
-   
+    string public tokenURI;
+
+    event TokenURIChanged(string tokenURI);
+
     /// @dev cotract to use when issuing a CC (Local Currency)
     /// @param _name string name for CC token that is created.
     /// @param _symbol string symbol for CC token that is created.
     /// @param _decimals uint8 percison for CC token that is created.
-    /// @param _totalSupply uint256 total supply of the CC token that is created. 
-    function ColuLocalCurrency(string _name, string _symbol, uint8 _decimals, uint256 _totalSupply) public {
-        require(_totalSupply != 0);     
+    /// @param _totalSupply uint256 total supply of the CC token that is created.
+    /// @param _tokenURI string the URI may point to a JSON file that conforms to the "Metadata JSON Schema".
+    function ColuLocalCurrency(string _name, string _symbol, uint8 _decimals, uint256 _totalSupply, string _tokenURI) public {
```

Comment:
Is it much more expensive to duplicate the contractor (and akk the functions that uses him, in the factories) so that ine will have `tokenURI` param and one that not (and will call the first one with `tokenURI = ''`?

Reply:
I checked this, turns out contracts can have only one constructor

Reply:
adding another `createCurrency`  functions is adds 56165 gas.
before: 3588323. after: 3532158.
It's around 1.5% of increase.

Reply:
about IssuanceFactory:
with new create functions it will take: 5248404 gas
without them:  5130855
So it adds 117549, about 2%